In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import statsmodels.formula.api as smf

In [3]:
spotify = pd.read_csv("Spotify Music Data.csv")
spotify

Unnamed: 0                                              title  \
0             1                                   Hey, Soul Sister   
1             2                               Love The Way You Lie   
2             3                                            TiK ToK   
3             4                                        Bad Romance   
4             5                               Just the Way You Are   
..          ...                                                ...   
598         599                Find U Again (feat. Camila Cabello)   
599         600      Cross Me (feat. Chance the Rapper & PnB Rock)   
600         601  No Brainer (feat. Justin Bieber, Chance the Ra...   
601         602    Nothing Breaks Like a Heart (feat. Miley Cyrus)   
602         603                                   Kills You Slowly   

               artist        top genre  year  bpm  nrgy  dnce  dB  live  val  \
0               Train       neo mellow  2010   97    89    67  -4     8   80   
1              Eminem  detroit hip hop  2010   87    93    75  -5    52   64   
2               Kesha        dance pop  2010  120    84    76  -3    29   71   
3           Lady Gaga        dance pop  2010  119    92    70  -4     8   71   
4          Bruno Mars              pop  2010  109    84    64  -5     9   43   
..                ...              ...   ...  ...   ...   ...  ..   ...  ...   
598       Mark Ronson        dance pop  2019  104    66    61  -7    20   16   
599        Ed Sheeran              pop  2019   95    79    75  -6     7   61   
600         DJ Khaled        dance pop  2019  136    76    53  -5     9   65   
601       Mark Ronson        dance pop  2019  114    79    60  -6    42   24   
602  The Chainsmokers       electropop  2019  150    44    70  -9    13   23   

     dur  acous  spch  pop  
0    217     19     4   83  
1    263     24    23   82  
2    200     10    14   80  
3    295      0     4   79  
4    221      2     4   78  
..   ...    ...   ...  ...  
598  176      1     3   75  
599  206     21    12   75  
600  260      7    34   70  
601  217      1     7   69  
602  213      6     6   67  

[603 rows x 15 columns]

### Training MLR

In [4]:
ols_object = smf.ols(formula = 'pop~bpm+nrgy+dnce+dB+live+val+dur+acous+spch', data = spotify)
mlr_spotify = ols_object.fit()
mlr_spotify.summary()


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    pop   R-squared:                       0.073
Model:                            OLS   Adj. R-squared:                  0.059
Method:                 Least Squares   F-statistic:                     5.170
Date:                Wed, 22 Feb 2023   Prob (F-statistic):           9.00e-07
Time:                        15:14:45   Log-Likelihood:                -2445.6
No. Observations:                 603   AIC:                             4911.
Df Residuals:                     593   BIC:                             4955.
Df Model:                           9                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     91.9428      8.605     10.684      0.000      75.042     108.844
bpm            0.0056      0.024      0.229      0.819      -0.042       0.053
nrgy          -0.1978      0.054     -3.657      0.000      -0.304      -0.092
dnce           0.0827      0.053      1.567      0.118      -0.021       0.186
dB             1.2858      0.258      4.989      0.000       0.780       1.792
live          -0.0515      0.045     -1.134      0.257      -0.141       0.038
val           -0.0056      0.033     -0.170      0.865      -0.070       0.059
dur           -0.0389      0.018     -2.201      0.028      -0.074      -0.004
acous         -0.0210      0.035     -0.598      0.550      -0.090       0.048
spch          -0.0057      0.079     -0.073      0.942      -0.161       0.150
==============================================================================
Omnibus:                      179.284   Durbin-Watson:                   0.497
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              524.025
Skew:                          -1.443   Prob(JB):                    1.62e-114
Kurtosis:                       6.539   Cond. No.                     4.19e+03
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 4.19e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

### Model significance
Is the overall regression significant at 5% level?

The overall regression is significant at the 5% level because the f-statistic is 9 * 10^-07, which is less than 0.05. This means that changes in popularity correlate with shifts in the predictor variables.

### Coefficient interpretation
Interpret the coefficient of every variable.

The coefficient for bpm is 0.0056. This means that there is a 0.0056 increase in bpm for every unit increase in popularity. 

The coefficient for nrgy is -0.1978. This means that there is a -0.1978 increase in nrgy for every unit increase in popularity. 

The coefficient for dnce is 0.0827. This means that there is a 0.0827 increase in dnce for every unit increase in popularity. 

The coefficient for dB is 1.2858. This means that there is a 1.2858 increase in dB for every unit increase in popularity. 

The coefficient for live is -0.0515. This means that there is a -0.0515 increase in live for every unit increase in popularity. 

The coefficient for val is -0.0056. This means that there is a -0.0056 increase in val for every unit increase in popularity. 

The coefficient for dur is -0.0389. This means that there is a -0.0389 increase in dur for every unit increase in popularity. 

The coefficient for acous is -0.0210. This means that there is a -0.0210 increase in acous for every unit increase in popularity. 

The coefficient for spch is -0.0057. This means that there is a -0.0057 increase in spch for every unit increase in popularity. 


### Variable significance
Report the $p$-values for all the predictor variables.

The coefficient for bpm is 0.819. This variable is not statistically significant because its p-value is higher than 0.05.

The coefficient for nrgy is 0.000. This variable is statically significant because its p-value is lower than 0.05.

The coefficient for dnce is 0.118. This variable is not statistically significant because its p-value is higher than 0.05.

The coefficient for dB is 0.000. This variable is statically significant because its p-value is lower than 0.05.


The coefficient for live is 0.257. This variable is not statistically significant because its p-value is higher than 0.05.

The coefficient for val is 0.865. This variable is not statistically significant because its p-value is higher than 0.05.

The coefficient for dur is 0.028. This variable is statically significant because its p-value is lower than 0.05.


The coefficient for acous is 0.550. This variable is not statistically significant because its p-value is higher than 0.05.

The coefficient for spch is 0.942. This variable is not statistically significant because its p-value is higher than 0.05.

